cache/transcoding.go

```diff
@@ -34,6 +36,29 @@ type SegmentInfo struct {
 	Outputs      []clients.OutputVideo // Information about the final transcoded outputs we've created
 }
 
+// Send "keepalive" callbacks to ensure the caller (Studio) knows transcoding is still ongoing and hasn't failed
+func (t *TranscodingCache) SendTranscodingHeartbeats(interval time.Duration, quit chan bool) {
+	for {
+		// Stop the infinite loop if we receive a quit message
+		select {
+		case <-quit:
+			return
+		default:
+		}
+
+		jobs := t.GetAll()
```

Comment:
Note: seems possible to have a race between this thread and transcode thread. We should expect on studio side to receive a heartbeat callback after transcode-complete callback. We should notify studio team about this quirk.

Reply:
Good point! I think Studio already has logic where they'll throw away messages if "Completion %" is less than one they've already received, which is quite a nice way of getting some thread safety around this stuff without them implementing a full on state machine